src/EnergyPlus/HybridEvapCoolingModel.hh

```diff
@@ -0,0 +1,346 @@
+			Real64 Minimum_Return_Air_Humidity_Ratio;
+			Real64 Maximum_Return_Air_Humidity_Ratio;
+			Real64 Minimum_Return_Air_Relative_Humidity;
+			Real64 Maximum_Return_Air_Relative_Humidity;
+			Real64 NormalizationReference;
+			Real64 Correction;
+			bool CMode::ValidPointer(int curve_pointer);
+			bool CMode::ValidateArrays(Array1D_string Alphas, Array1D_string cAlphaFields, Array1D< Real64 > Numbers, Array1D_string cNumericFields, std::string cCurrentModuleObject);
+			bool CMode::ParseMode(Array1D_string Alphas, Array1D_string cAlphaFields, Array1D< Real64 > Numbers, Array1D_string cNumericFields, Array1D<bool>  lAlphaBlanks, std::string cCurrentModuleObject);
+			bool CMode::ParseMode0(Array1D_string Alphas, Array1D_string cAlphaFields, Array1D< Real64 > Numbers, Array1D_string cNumericFields, Array1D<bool>  lAlphaBlanks, std::string cCurrentModuleObject);
+		
+			void CMode::InitializeCurve( int curveType, int CurveID);
+			Real64 CMode::CalculateCurveVal(Real64 X_0, Real64 X_1, Real64 X_2, Real64 X_3, Real64 X_4, Real64 X_5, Real64 X_6, int mode_number, int curve_ID);
+			bool CMode::InitializeOSAFConstraints(Real64 minOSAF, Real64 maxOSAF);
+			bool CMode::InitializeMsaRatioConstraints(Real64 minMsa, Real64 maxMsa);
+			bool CMode::InitializeOutsideAirTemperatureConstraints(Real64 min, Real64 max);
+			bool CMode::InitializeOutsideAirHumidityRatioConstraints(Real64 min, Real64 max);
+			bool CMode::InitializeOutsideAirRelativeHumidityConstraints(Real64 min, Real64 max);
+			bool CMode::InitializeReturnAirTemperatureConstraints(Real64 min, Real64 max);
+			bool CMode::InitializeReturnAirHumidityRatioConstraints(Real64 min, Real64 max);
+			bool CMode::InitializeReturnAirRelativeHumidityConstraints(Real64 min, Real64 max); 
+			bool CMode::GenerateSolutionSpace(Real64 ResolutionMsa, Real64 ResolutionOSA);
+			bool CMode::MeetsOAEnvConstraints(Real64 Tosa, Real64 Wosa, Real64 RHos);
+		private:
+		
+		};
+		
+		class CSetting
+		{
+		public:
+			CSetting() :Runtime_Fraction(0), Mode(0), Outside_Air_Fraction(0), Supply_Air_Mass_Flow_Rate(0), Supply_Air_Ventilation_Volume(0), Supply_Air_Mass_Flow_Rate_Ratio(0),
+				SupplyAirTemperature(0), Mixed_Air_Temperature(0), SupplyAirW(0), Mixed_Air_W(0), TotalSystem(0), SensibleSystem(0), LatentSystem(0), TotalZone(0), SensibleZone(0), LatentZone(0), ElectricalPower(IMPLAUSIBLE_POWER), pMode(NULL){}
+			Real64 Runtime_Fraction;
+			Real64 Mode;
+			Real64 Outside_Air_Fraction;
+			Real64 Supply_Air_Mass_Flow_Rate;
+			Real64 Supply_Air_Ventilation_Volume;
+			Real64 Supply_Air_Mass_Flow_Rate_Ratio;
+			Real64 SupplyAirTemperature;
+			Real64 Mixed_Air_Temperature;
+			Real64 SupplyAirW;
+			Real64 Mixed_Air_W;
+			Real64 TotalSystem;
+			Real64 SensibleSystem;
+			Real64 LatentSystem;
+			Real64 TotalZone;
+			Real64 SensibleZone;
+			Real64 LatentZone;
+			Real64 ElectricalPower;
+			// add other fuels, or change name to be fuel
+			CMode* pMode;
+			CSetting& operator=(CSetting other)
```

Comment:
I know @mbadams5 is busy, so I'll say that this makes me pretty nervous. I'd definitely rather not see the `using std::swap`. I don't think we have this sort of thing anyplace else in the codebase, so my vote would be to find another way to do this or at least have a broader discussion of how we approach things.